### PR TITLE
Allow users to add course choices to an application

### PR DESCRIPTION
**Pull request by tvararu**
_PR description by @tijmenb_

### Context

This PR implements the starting point for choosing a course. It's not finished, but is functional.

Bits that will be implemented in future PRs are [documented in the checklist for this ticket](https://trello.com/c/HaxxM4vH/196-selecting-a-course-%F0%9F%8F%88-kickoff).

### Changes proposed in this pull request

The bare minimum to create a course choice.

### Guidance to review

See if it makes sense.

### Screenshots

Screenshot 2019-11-01 at 18 24 17
Screenshot 2019-11-01 at 18 24 22
Screenshot 2019-11-01 at 18 24 27
Screenshot 2019-11-01 at 18 24 34
Screenshot 2019-11-01 at 18 24 42
Screenshot 2019-11-01 at 18 24 49
### Link to Trello card

[196 - Selecting a course](https://trello.com/c/HaxxM4vH/196-selecting-a-course-🏈-kickoff)

### CODE REVIEW

**Comment by tijmenb** [comment]
I've updated the PR and PR description, I think it's good to be merged.

File: app/views/candidate_interface/course_choices/have_you_chosen.html.erb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<% content_for :title, page_title('have_you_chosen') %>
+<% content_for :before_content, govuk_back_link_to(candidate_interface_application_form_path, 'Back to application') %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <%= form_with model: @choice, url: candidate_interface_course_choices_choose_path, builder: GOVUKDesignSystemFormBuilder::FormBuilder do |f| %>
+      <%= f.govuk_error_summary %>
+
+      <div class="govuk-!-margin-top-6">
+        <%= f.govuk_radio_buttons_fieldset :choice, legend: { size: 'xl', text: t('page_titles.have_you_chosen') } do %>
```
Comment (chubberlisk) [inline app/views/candidate_interface/course_choices/have_you_chosen.html.erb]: Another smol thing:

Ideally there should be a larger margin between the heading and the radio buttons like the prototype. Couldn't seem to pass through the CSS class for the `govuk-!-margin-bottom-8` overwrite though.

File: app/views/candidate_interface/course_choices/have_you_chosen.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<% content_for :title, page_title('have_you_chosen') %>
+<% content_for :before_content, govuk_back_link_to(candidate_interface_application_form_path, 'Back to application') %>
```
Comment (chubberlisk) [inline app/views/candidate_interface/course_choices/have_you_chosen.html.erb]: Thinking maybe this should go to `candidate_interface_course_choices_index_path` and potentially the others as well.

File: app/views/candidate_interface/course_choices/have_you_chosen.html.erb (hunk 1 above)
Comment (tijmenb) [reply app/views/candidate_interface/course_choices/have_you_chosen.html.erb]: Hmm, I'm getting a little confused about the layout. I've added this as a separate checklist item to https://trello.com/c/HaxxM4vH/196-selecting-a-course-%F0%9F%8F%88-kickoff to make sure all the form pages look the same.

File: app/views/candidate_interface/application_form/show.html.erb [hunk 3]
```diff
@@ -12,7 +12,9 @@
     <% unless @application_form_presenter.application_choices_added? %>
       <p class="govuk-body"><%= t('application_form.courses') %></p>
     <% end %>
-    <%= govuk_button_link_to 'TODO: Add course', '#' %>
+    <p class="govuk-body">
+      <%= govuk_link_to 'Course choices', candidate_interface_course_choices_index_path %>
+    </p>
```
[the remark this answers is not in the document]
Comment (tijmenb) [reply app/views/candidate_interface/application_form/show.html.erb]: Done. Also added "Allow the section to be marked as complete" to the checklist on the card.

File: app/views/candidate_interface/course_choices/have_you_chosen.html.erb (hunk 2 above)
Comment (tijmenb) [reply app/views/candidate_interface/course_choices/have_you_chosen.html.erb]: Done!

**Pull request merged**